embedding/embedlite/embedshared/nsWindow.cpp

```diff
@@ -276,11 +285,65 @@ layers::LayersId nsWindow::GetRootLayerId() const
   return mCompositorSession ? mCompositorSession->RootLayerTreeId() : layers::LayersId{0};
 }
 
-void nsWindow::SetContentController(mozilla::layers::GeckoContentController *aController)
+#ifdef MOZ_LOGGING
+static void LogControllerStack(std::list<EmbedContentController *> &controllers)
+{
+  int pos = 0;
+  for (std::list<EmbedContentController *>::const_iterator it = controllers.begin(); it != controllers.end(); ++it) {
+    LOGT("Stack controller %d : %u", pos, (*it)->GetUniqueID());
+    ++pos;
+  }
+}
+#endif // MOZ_LOGGING
+
+inline static bool ControllersEqual(const EmbedContentController *aController1, const EmbedContentController *aController2)
+{
+  // Compare unique ids, but fallback to pointer comparison if id is zero
+  return (aController1->GetUniqueID() != 0 && (aController1->GetUniqueID() == aController2->GetUniqueID()))
+      || (aController1 == aController2);
```

Comment:
What's reason for the fallback? If unique ids are not matching then this should return false, right?

Reply:
Let's leave it like it is.